cli/formatter/formatter.go

```diff
@@ -0,0 +1,778 @@
+			}
+		} else {
+			// unpack array with array(s)
+			for _, node := range renderBatch {
+				bytesNode, err := json.Marshal(convertMapIntrfToMapStr(node))
+				if err != nil {
+					return err
+				}
+				iterableNode, err := decodeYamlArr(string(bytesNode))
+				if err != nil {
+					return err
+				}
+				if err := renderArrs(iterableNode, formatterOpts); err != nil {
+					return err
+				}
+			}
+		}
+
+		return nil
+	}
+
+	return nil
+}
+
+// RenderTables renders tables for table/ttable output formats.
+func RenderTables(inputYaml string, formatterOpts *Opts) {
+	// handle empty input from remote console
+	if inputYaml == "---\n- \n...\n" || inputYaml == "---\n-\n...\n" {
+		inputYaml = "--- ['']\n...\n"
+	}
+	if renderNodes, err := decodeYamlArr(inputYaml); err != nil {
+		log.Errorf("unexpected server response: not yaml array, cannot render tables: %v", err)
+		fmt.Println(inputYaml)
+	} else {
+		var renderBatchs = make([][]renderNode, len(renderNodes))
+		var batchPointer = 0
+		renderBatchs[batchPointer] = append(renderBatchs[batchPointer], renderNodes[0])
+
+		for i := 0; i < len(renderNodes)-1; i++ {
+			if isRenderNodesTypesEqual(renderNodes[i], renderNodes[i+1]) {
+				renderBatchs[batchPointer] = append(renderBatchs[batchPointer], renderNodes[i+1])
+			} else {
+				batchPointer = batchPointer + 1
+				renderBatchs[batchPointer] = append(renderBatchs[batchPointer], renderNodes[i+1])
+			}
+		}
+
+		for _, batch := range renderBatchs {
+			if len(batch) != 0 {
+				if err := parseRenderBatch(batch, formatterOpts); err != nil {
+					log.Errorf("unexpected internal parser result, cannot render tables: %v", err)
+					fmt.Println(inputYaml)
+				}
+				if formatterOpts.NoGraphics {
+					fmt.Println()
+				}
+			}
+		}
+	}
+}
+
+// luaEncodeElement encodes element to lua string.
+func luaEncodeElement(elem interface{}) string {
+	switch elem.(type) {
+	case map[interface{}]interface{}:
+		res := "{"
+		for k, v := range elem.(map[interface{}]interface{}) {
+			if v != nil {
+				if reflect.TypeOf(v).String() == "string" {
+					v = "\"" + fmt.Sprintf("%v", v) + "\""
+				}
+			} else {
+				v = "nil"
+			}
+
+			if reflect.TypeOf(k).String() == "string" {
+				res = res + fmt.Sprintf("%v", k) + " = " + luaEncodeElement(v) + ","
+			} else {
+				res = res + "[" + fmt.Sprintf("%v", k) + "] = " + luaEncodeElement(v) + ","
+			}
+		}
+		return res + "}"
+	case []interface{}:
+		res := "{"
+		for k, v := range elem.([]interface{}) {
+			if v != nil {
+				if reflect.TypeOf(v).String() == "string" {
+					v = "\"" + fmt.Sprintf("%v", v) + "\""
+				}
+			} else {
+				v = "nil"
+			}
+			if k < len(elem.([]interface{}))-1 {
+				res = res + luaEncodeElement(v) + ", "
+			} else {
+				res = res + luaEncodeElement(v)
+			}
+		}
+		return res + "}"
+	default:
+		if elem == nil {
+			return "nil"
+		}
+		return fmt.Sprintf("%v", elem)
+	}
+}
+
+// RenderLua renders lua string from yaml string.
+func RenderLua(inputYaml string) {
```

Comment:
I suggest do not print the result to the stdout, but return as a string [2]
```suggestion
func RenderLua(inputYaml string) string {
```

Reply:
Done